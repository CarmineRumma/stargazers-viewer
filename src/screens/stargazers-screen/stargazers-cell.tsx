import React from 'react';
import {Appearance, Image, ListRenderItemInfo, Text, TouchableHighlight, View} from 'react-native';
import {RepositoryItem} from '@stargazers/services/user-service/user-service.types';
import {Colors, Fonts} from '@stargazers/theme';
import {styles} from '@stargazers/screens/repositories-screen/repository-cell.styles';
import SvgArrowRight from '@stargazers/assets/Svg.ArrowRight';
import moment from 'moment';
import useColorScheme from '@stargazers/hooks/useColorScheme';
//import FastImage from 'react-native-fast-image';

const getColorScheme = Appearance.getColorScheme;
const colorScheme = getColorScheme()!;

export interface CategoryCellProps extends ListRenderItemInfo<RepositoryItem> {}

export const StargazersCell: React.FC<CategoryCellProps> = ({
  item,
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  index,
  separators,
}) => {
  const colorScheme = useColorScheme();

  return (
    <>
      <TouchableHighlight
        // @ts-ignore
        style={{
          ...styles.cell,
          borderBottomColor: Colors[colorScheme].cellBorderBottomColor,
          backgroundColor: Colors[colorScheme].background,
        }}
        underlayColor={Colors[colorScheme].cellHighlight}
        onShowUnderlay={separators.highlight}
        onHideUnderlay={separators.unhighlight}
        onPress={() => console.log(item.name)}>
        <View style={styles.cellWrapper}>
          <View
            style={{
              ...styles.imageWrapper,
            }}>
            <Image
              source={require('../../assets/repo-icon.png')}
              style={{width: 24, height: 24, alignSelf: 'center', alignItems: 'center'}}
            />
          </View>
          <View style={styles.textsContainer}>
            <Text
              style={{
                ...styles.productName,
                color: Colors[colorScheme].tint,
              }}
              numberOfLines={2}
              ellipsizeMode="tail">
              {item.name}
            </Text>

            <View style={styles.updatedAtWrapper}>
              <Text
                style={{
                  ...styles.updatedAt,
                  color: Colors[colorScheme].tint,
                }}>
                {moment(item.updated_at).format('MMMM d')},{' '}
                <Text style={{fontWeight: 'bold'}}>{moment(item.updated_at).format('HH:mm')}</Text>
              </Text>
            </View>
            <View style={styles.counterRowsWrapper}>
              <Image
                source={require('../../assets/star-icon.png')}
                style={styles.starCounterImage}
              />
              <Text style={styles.starCounterText}>{item.stargazers_count}</Text>
            </View>
          </View>
          <View style={styles.rightColumn}>
            <SvgArrowRight
              size={20}
              style={{
                color: Colors[colorScheme].tint,
              }}
            />
          </View>
        </View>
      </TouchableHighlight>
    </>
  );
};
