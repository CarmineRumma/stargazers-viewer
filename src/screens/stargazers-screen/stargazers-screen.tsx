import {StargazersScreenProps} from '@stargazers/navigators/navigator.types';
import React from 'react';
import {View, Text} from 'react-native';

export const StargazersScreen = ({route, navigation}: StargazersScreenProps) => {
  return (
    <View>
      <Text>Stargeazers</Text>
    </View>
  );
};
