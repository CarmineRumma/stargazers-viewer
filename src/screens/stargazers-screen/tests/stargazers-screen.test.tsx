import React from 'react';

import renderer from 'react-test-renderer';
import {StargazersScreen} from '../stargazers-screen';

jest.mock('react-native-vector-icons/Feather', () => 'Feather');
const goBack = jest.fn();
const navigate = jest.fn();
const setOptions = jest.fn();
const addListener = jest.fn();
const mockRoute = {
  params: {
    owner: 'test',
    repository: {name: 'test'},
    stargazers: [],
  },
};
const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
      setOptions: jest.fn(),
    }),
  };
});
describe('HomeScreen', () => {
  it('should render correctly', () => {
    const {toJSON} = renderer.create(
      <StargazersScreen
        //@ts-ignore
        navigation={{goBack, navigate, setOptions, addListener}}
        //@ts-ignore
        route={mockRoute}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
