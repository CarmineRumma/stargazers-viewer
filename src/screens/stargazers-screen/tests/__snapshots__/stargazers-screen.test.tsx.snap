// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "height": 50,
        "justifyContent": "center",
        "paddingHorizontal": 10,
        "paddingVertical": 4,
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Search stargazer.."
      placeholderTextColor="#FFFFFF"
      style={
        Object {
          "backgroundColor": "#414757",
          "borderColor": "#333",
          "borderRadius": 25,
          "color": "#FFFFFF",
          "fontSize": 14,
          "height": 40,
          "paddingHorizontal": 15,
          "width": "100%",
        }
      }
    />
  </View>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListEmptyComponent={[Function]}
    ListHeaderComponent={[Function]}
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    removeClippedSubviews={true}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    style={Object {}}
    viewabilityConfigCallbackPairs={Array []}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "paddingHorizontal": 14,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#414757",
                "fontFamily": "HelveticaNeue-Medium",
                "fontSize": 15,
                "padding": 3,
              }
            }
          >
            0 stargazers
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "paddingTop": 333.5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#383B42",
              "fontFamily": "Montserrat-SemiBold",
              "fontSize": 26,
              "maxWidth": "90%",
              "textAlign": "center",
            }
          }
        >
          no stargazer...yet
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
